OpenRA.Mods.Common/Traits/SupportPowers/SupportPowerManager.cs

```diff
@@ -272,6 +278,26 @@ public virtual string TooltipTimeTextOverride()
 		{
 			return null;
 		}
+
+		bool HasSufficientFunds(SupportPower power, bool activate = false)
+		{
+			if (power.Info.Cost != 0)
+			{
+				var player = Manager.Self;
+				var pr = player.Trait<PlayerResources>();
+				if (pr.Cash + pr.Resources < power.Info.Cost)
+				{
+					Game.Sound.PlayNotification(player.World.Map.Rules, player.Owner, "Speech",
+						pr.Info.InsufficientFundsNotification, player.Owner.Faction.InternalName);
+					return false;
+				}
+
+				if (activate)
+					pr.TakeCash(power.Info.Cost);
```

Comment:
This pattern here looks very weird to me: A method with a name that implies it only checks if it there is enough cash actually removes cash. Wouldn't it make more sense to call this line directly from within `Power.Activate`?